File: array/sort0s1s2s.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int low=0;
        int mid=0;
        int high=nums.size()-1;

        while(mid<=high){
            int temp;
            if(nums[mid]==0){
            temp=nums[low];
            nums[low]=nums[mid];
            nums[mid]=temp;

            low++;
            mid++;
            }
            else if(nums[mid]==1){
                mid++;
            }
            else if(nums[mid]==2){
                temp=nums[high];
                nums[high]=nums[mid];
                nums[mid]=temp;

                high--;
            }
        }

    }
};

File: array/sort0s1s2s.js
var sortColors = function(nums) {
    let l=0;
    let m=0;
    let h= nums.length-1;

    while(m<=h){
        let temp;

        if(nums[m]===0){
            temp=nums[l];
            nums[l]=nums[m];
            nums[m]=temp;

            l++;
            m++;
        }
        else if(nums[m]===1){
            m++;
        }
        else if(nums[m]===2){
            temp=nums[h];
            nums[h]=nums[m];
            nums[m]=temp;

            h--;
        }
    }
};
    
